lib/inc/internal/config_impl_util.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace hocon {
+
+    class config_impl_util {
+    public:
+        static bool equals_handling_null(std::string const& a, std::string const& b);
```

Comment:
Since this is only used for strings and vectors, I'm not sure it's needed. They'll never be null because we're using references, so we can just do `a == b` and call it a day.